experimental/traffic-portal/src/app/shared/tp-header/tp-header.component.ts

```diff
@@ -37,17 +37,55 @@ export class TpHeaderComponent implements OnInit {
 
 	public hidden = false;
 
+	// Will try to display each of these navs on the header, space allowing.
+	public horizNavs: Array<HeaderNavigation> = new Array<HeaderNavigation>();
+	// Navs that are not directly displayed on the header.
+	public vertNavs: Array<HeaderNavigation> = new Array<HeaderNavigation>();
+
 	/**
 	 * Angular lifecycle hook
 	 */
 	public ngOnInit(): void {
+		this.headerSvc.addHorizontalNav({
+			routerLink: "/core",
+			text: "Home",
+			type: "anchor",
+		}, "home");
+		this.headerSvc.addHorizontalNav({
+			routerLink: "/core/users",
+			text: "Users",
+			type: "anchor",
+			visible: () => this.hasPermission("USER:READ"),
+		}, "Users");
+		this.headerSvc.addHorizontalNav({
+			routerLink: "/core/servers",
+			text: "Servers",
+			type: "anchor",
+			visible: () => this.hasPermission("SERVER:READ"),
+		}, "Servers");
+		this.headerSvc.addVerticalNav({
+			routerLink: "/core/me",
+			text: "Profile",
+			type: "anchor"
+		}, "Profile");
+		this.headerSvc.addVerticalNav({
+			click: async () => this.logout(),
+			text: "Logout",
+			type: "anchor"
+		}, "Logout");
+
 		this.headerSvc.headerTitle.subscribe(title => {
 			this.title = title;
 		});
-
 		this.headerSvc.headerHidden.subscribe(hidden => {
 			this.hidden = hidden;
 		});
+		this.headerSvc.horizontalNavsUpdated.subscribe(navs => {
+			this.horizNavs = navs;
+		});
+		this.headerSvc.verticalNavsUpdated.subscribe(navs => {
+			this.vertNavs = navs;
+		});
```

Comment:
Instead of having the component attempt to re-insert these default links every time it renders - which will overwrite my overrides of default links - why not just have the service own them, which is a singleton by definition and won't overrwrite changes made through the service's API?

Reply:
Will do. For the record, It only redid the inserts on the initial creation of the component, which should only happen once.